File: lib/languages/typescript/type-annotations.ts
// Basic type annotations
const name = "Alice"
const age = 30
const height = 1.75
const isActive = true

// Function type annotations
function greet(name: string): string {
  return `Hello, ${name}`
}

// Type aliases
type UserId = number
type UserName = string

// Interface definitions
interface User {
  id: UserId
  name: UserName
  age: number
}

// Generic type annotations
type Result<T, E> = { ok: true; value: T } | { ok: false; error: E }

// Using generic types
const success: Result<number, string> = { ok: true, value: 42 }
const failure: Result<number, string> = { ok: false, error: "Something went wrong" }

// Complex type operations
type Partial<T> = { [P in keyof T]?: T[P] }
type ReadOnly<T> = { readonly [P in keyof T]: T[P] }

// Type assertions
const element = document.getElementById("root") as HTMLDivElement

File: lib/languages/rust/type-annotations.rs
// Basic type annotations
let name: String = String::from("Alice");
let age: u32 = 30;
let height: f64 = 1.75;
let is_active: bool = true;

// Function type annotations
fn greet(name: &str) -> String {
   format!("Hello, {}!", name)
}

// Type aliases
type UserId = u64;
type UserName = String;

// Struct definitions
struct User {
   id: UserId,
   name: UserName,
   age: u32,
}

// Enum definitions
enum Result<T, E> {
   Ok(T),
   Err(E),
}

// Using generic types
let success: Result<i32, String> = Result::Ok(42);
let failure: Result<i32, String> = Result::Err(String::from("Something went wrong"));

// Trait bounds
fn print_id<T: std::fmt::Display>(id: T) {
   println!("ID: {}", id);
}

